web/client-api/src/main/java/io/deephaven/web/client/api/tree/JsTreeTable.java

```diff
@@ -748,7 +756,40 @@ public void setExpanded(Object row, boolean isExpanded) {
             throw new IllegalArgumentException("row parameter must be an index or a row");
         }
 
-        r.appendKeyData(keyTableData, isExpanded);
+        r.appendKeyData(keyTableData, action);
+        if (keyTable != null) {
+            keyTable.then(t -> {
+                t.close();
+                return null;
+            });
+            keyTable = null;
+        }
+        replaceSubscription(RebuildStep.HIERARCHICAL_TABLE_VIEW);
```

Comment:
This is used in `setExpanded`, `expandAll`, and `collapseAll`, you could refactor it into it's own method, e.g.
```
private void replaceKeyTable() {
    if (keyTable != null) {
        keyTable.then(t -> {
            t.close();
            return null;
        });
        keyTable = null;
    }
    replaceSubscription(RebuildStep.HIERARCHICAL_TABLE_VIEW);
} 
```

And then here just call that method:
```suggestion
        replaceKeyTable();
```